src/IR/LowerAst.hs

```diff
@@ -133,6 +137,16 @@ lowerExpr (A.IfElse c t e) k = I.Match cond Nothing [tArm, eArm] (k I.untyped)
 lowerExpr (A.OpRegion _ _) _ = error "Should already be desugared"
 lowerExpr A.NoExpr         k = I.Lit I.LitEvent (k I.untyped)
 
+-- | Lower an A.Pat into an I.Alt
+lowerAlt :: A.Pat -> I.Alt
+lowerAlt A.PatWildcard  = I.AltDefault
+lowerAlt (A.PatId  _  ) = error "I.Alt for A.PatID not implemented yet"
+lowerAlt (A.PatLit l  ) = I.AltLit $ lowerLit l
+lowerAlt (A.PatTup _  ) = error "I.Alt for A.PatTup not implemented yet"
+lowerAlt (A.PatCon _ _) = error "No way to lower A.DataCons to I.DataCons yet"
+lowerAlt (A.PatAnn _ p) = lowerAlt p
+lowerAlt (A.PatAs  _ p) = lowerAlt p
```

Comment:
You should throw an error here; something of the form `match x { a@b => { e }` should be desugared to `match x { b => { let a = b ; e }`. But that should all be part of more extensive desugaring, and shouldn't be handled here for now.